tests/compas_timber/test_connections_solvers.py

```diff
@@ -0,0 +1,361 @@
+import pytest
+from compas_timber.elements import Beam
+from compas_timber.connections import ConnectionSolver
+from compas_timber.connections import JointTopology
+from compas.geometry import Point
+from compas_timber.elements import Plate
+from compas_timber.connections import PlateConnectionSolver
+from compas.geometry import Polyline
+from compas_timber.utils import distance_segment_segment
+from compas.tolerance import TOL
+
+
+def flip_beam(beam):
+    return Beam.from_endpoints(beam.centerline.end, beam.centerline.start, beam.width, beam.height)
+
+
+@pytest.fixture
+def i_beams():
+    first_beam = Beam.from_endpoints(Point(0, 0, 0), Point(10, 0, 0), 1, 1)
+    connecting_beams = [
+        Beam.from_endpoints(Point(-10, 0, 0), Point(0, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(10, 0, 0), Point(20, 0, 0), 1, 1),
+    ]
+    return first_beam, connecting_beams
+
+
+@pytest.fixture
+def overlap_i_beams():
+    first_beam = Beam.from_endpoints(Point(0, 0, 0), Point(10, 0, 0), 1, 1)
+    connecting_beams = [
+        Beam.from_endpoints(Point(0.01, 0, 0), Point(10, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(0, 0, 0), Point(9.99, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(0, 0, 0), Point(10.01, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(-0.01, 0, 0), Point(10, 0, 0), 1, 1),
+    ]
+    return first_beam, connecting_beams
+
+
+@pytest.fixture
+def i_beams_with_tolerance():
+    first_beam = Beam.from_endpoints(Point(0, 0, 0), Point(10, 0, 0), 1, 1)
+    connecting_beams = [
+        Beam.from_endpoints(Point(-10, 0, 0), Point(-0.01, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(10.01, 0, 0), Point(20, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(-10, 0, 0), Point(0.01, 0, 0), 1, 1),
+        Beam.from_endpoints(Point(9.99, 0, 0), Point(20, 0, 0), 1, 1),
+    ]
+    return first_beam, connecting_beams
+
+
+def test_find_topology_i_beams(i_beams):
+    for beam_a in [i_beams[0], flip_beam(i_beams[0])]:
+        for bb in i_beams[1]:
+            for beam_b in [bb, flip_beam(bb)]:
```

Comment:
unittests should read very simple, perhaps this could be flattened with `parametrize` and some hard-coded input?